packages/date-picker/src/DatePicker/DatePicker.tsx

```diff
@@ -190,6 +263,36 @@ export const DatePicker: React.VFC<DatePickerProps> = ({
     }
   }
 
+  useEffect(() => {
```

Comment:
By validating after the first render has been committed, the validation error won’t appear in the server rendered output. Not sure if that’s something that matters?

Reply:
The initial render validation is more of an edge case since the user has yet to interact with the datepicker. It is for the situation when the api has given an invalid or disabled date, so we communicate to the user that there may be something they need to fix, but nothing more than.

Unless you can think of an example where you'd want us to handle it differently?